src/components/OfflineIndicator.js

```diff
@@ -36,20 +36,22 @@ const defaultProps = {
     style: [],
 };
 
-const setStyles = (containerStyles, isSmallScreenWidth) => {
+const getStyles = (containerStyles, isSmallScreenWidth, styles) => {
     if (containerStyles.length) {
         return containerStyles;
     }
     return isSmallScreenWidth ? styles.offlineIndicatorMobile : styles.offlineIndicator;
 };
```

Comment:
Let's bring this to inside `OfflineIndicator` and memoize it.

Reply:
Let's also not pass styles as a prop. The function is only used once, let's construct the styles directly

```js
    const containerStyle = useMemo(() => {
        if (props.containerStyles.length) {
            return props.containerStyles;
        }
        return props.isSmallScreenWidth ? styles.offlineIndicatorMobile : styles.offlineIndicator;
    }, [styles, props.containerStyles, props.isSmallScreenWidth]);
```